driver/src/main/java/org/neo4j/driver/Session.java

```diff
@@ -87,67 +88,169 @@
+     * @param <T>      the return type of the given unit of work.
+     * @return a result as returned by the given unit of work.
+     */
+    <T> T executeRead( TransactionCallback<T> callback, TransactionConfig config );
+
     /**
      * Execute a unit of work in a managed {@link AccessMode#WRITE write} transaction.
      * <p>
-     * This transaction will automatically be committed unless an exception is
-     * thrown during query execution or by the user code.
+     * This transaction will automatically be committed unless an exception is thrown during query execution or by the user code.
      * <p>
-     * Managed transactions should not generally be explicitly committed (via
-     * {@link Transaction#commit()}).
+     * Managed transactions should not generally be explicitly committed (via {@link Transaction#commit()}).
      *
      * @param work the {@link TransactionWork} to be applied to a new write transaction.
-     * @param <T> the return type of the given unit of work.
+     * @param <T>  the return type of the given unit of work.
      * @return a result as returned by the given unit of work.
+     * @deprecated superseded by {@link #executeWrite(TransactionCallback)}.
      */
+    @Deprecated
     <T> T writeTransaction( TransactionWork<T> work );
 
     /**
-     * Execute a unit of work in a managed {@link AccessMode#WRITE write} transaction
-     * with the specified {@link TransactionConfig configuration}.
+     * Execute a unit of work as a single managed transaction with {@link AccessMode#WRITE write} access mode and retry behaviour.
      * <p>
-     * This transaction will automatically be committed unless an exception is
-     * thrown during query execution or by the user code.
+     * The driver will attempt committing the transaction when the provided unit of work completes successfully. A user initiated failure of the unit of work
+     * will result in rollback attempt.
      * <p>
-     * Managed transactions should not generally be explicitly committed (via
-     * {@link Transaction#commit()}).
+     * The provided unit of work should not return {@link Result} object.
      *
-     * @param work the {@link TransactionWork} to be applied to a new write transaction.
+     * @param callback the callback representing the unit of work.
+     * @param <T>      the return type of the given unit of work.
+     * @return a result as returned by the given unit of work.
+     */
+    default <T> T executeWrite( TransactionCallback<T> callback )
+    {
+        return executeWrite( callback, TransactionConfig.empty() );
+    }
+
+    /**
+     * Execute a unit of work as a single managed transaction with {@link AccessMode#WRITE write} access mode and retry behaviour.
+     * <p>
+     * The driver will attempt committing the transaction when the provided unit of work completes successfully. A user initiated failure of the unit of work
+     * will result in rollback attempt.
```

Comment:
```suggestion
     * The driver will attempt committing the transaction when the provided unit of work completes successfully. Any runtime exception thrown by of the unit of work
     * will result in a rollback attempt.
```

Reply:
Thanks. I have made this part a bit more detailed.